streamr/client/client.py

```diff
@@ -0,0 +1,398 @@
+
+        def resendResponseNoResend(msg):
+            if self.subById.get(msg.payload.subId,None):
+                self.subById[msg.payload.subId].emit('no_resend',msg.payload)
+            else:
+                logger.debug('resent: Subscrption %s is gone already'%(msg.payload.subId))
+        self.connection.on('ResendResponseNoResend', resendResponseNoResend)
+    
+        def connectedListener():
+            logger.debug('Connected')
+            for key in self.subsByStream.keys():
+                subs = self.subsByStream[key]
+                for sub in subs:
+                    if sub.getState() != Subscription.State.subscribed:
+                        self.__resendAndSubscribe(sub)
+
+            publishQueueCopy = self.publishQueue
+            self.publishQueue = []
+            for element in publishQueueCopy:
+                self.publish(*element)
+            self.emit('connected')
+        self.connection.on('connected', connectedListener)
+
+
+        def disconnectedListener():
+            logger.debug('Disconnected')
+
+            for k in self.subsByStream.keys():
+                subs = self.subsByStream[k]
+                for sub in subs:
+                    sub.setState(Subscription.State.unsubscribed)
+
+            self.emit('disconnected')
+        self.connection.on('disconnected', disconnectedListener)
+
+        def errorResponse(err):
+            errorObject = Exception(err.payload.error)
+            self.emit('error',errorObject)
+            logger.error(errorObject)
+        self.connection.on('ErrorResponse',errorResponse)
+
+        def error(err):
+            if isinstance(err,InvalidJsonError):
+                subs = self.subsByStream[err.streamId]
+                for sub in subs:
+                    sub.handleError(err)
+            else:
+                logger.error(err.with_traceback(err.__traceback__))
+        self.connection.on('error',error)
+
+
+
+    def __autoUpdateSessionToken(self):
+        self.sessionToken = getSeToken_APIKey(self.options['apiKey'])
+        t = threading.Timer(self.options.get('sessionTokenRefreshInterval',7200),self.__autoUpdateSessionToken)
+        self.sessionThread = t
+        t.start()
+
+    def __addSubscription(self,sub):
+        self.subById[sub.id] = sub
+        self.subsByStream[sub.streamId].append(sub)
+
+    def __removeSubscription(self,sub):
+        if sub.id in self.subById.keys():
+            self.subById.pop(sub.id)
+        subs = self.subsByStream.get(sub.streamId, [])
+        if  len(subs) != 0:
+            for i in sorted(range(len(subs)),reverse=True):
+                if subs[i].id == sub.id:
+                    subs.pop(i)
+            self.subsByStream[sub.streamId] = subs
+    
+    def publish(self,objectOrId,data,apiKey = None):
+        if apiKey == None:
+            apiKey = self.options['auth']['apiKey']
+            
+        streamId = objectOrId.streamId
```

Comment:
I would suggest adding option to publish by providing simple streamId string, in addition to getting it from object attribute.
```
if hasattr(objectOrId, 'streamId'):
    streamId = objectOrId.streamId
else:
    streamId = objectOrId
```

Reply:
Thanks for your suggestion, I will revise it later.